### PR TITLE
Add `--class-balance` flag to `train` CLI

### PR DESCRIPTION
**Pull request by shihchengli**
## Description
I have added the `--class-balance` flag and a corresponding test for the train CLI.

## Checklist
- [x] linted with flake8?
- [x] (if appropriate) unit tests added?

### CODE REVIEW

**Review by akshatzalte — COMMENTED** [review]
a few minor edits, will approve as soon as those are addressed!

File: chemprop/cli/train.py [hunk 1]
```diff
@@ -1031,8 +1036,16 @@ def main(args):
...
-            train_dset, args.batch_size, args.num_workers, seed=args.data_seed
+            train_dset,
+            args.batch_size,
+            args.num_workers,
+            class_balance=args.class_balance,
+            seed=args.data_seed,
         )
+        if args.class_balance:
+            logger.debug(
+                f"With `--class-balance`, effective train size = {len(train_loader.sampler)}"
```
[the remark this answers is not in the document]
Comment (shihchengli) [reply chemprop/cli/train.py:1047]: Effective size means the number of data points used to train the model after balancing the positive and negative labels. As the number of positive labels and that of negative labels would be different thus the training size would be adjusted.

**Review by akshatzalte — APPROVED** [review]
